### ISSUE TITLE
Get-AzurRMDNSZone cmdlet not outputting correct results when using where-object

### ISSUE THREAD

**Issue by animetauren** [Azure/azure-powershell#4203]
### Cmdlet(s)
Get-AzureRmDnsZone

### PowerShell Version: 
5.1.15063.413

### Module Version
AzureRM.Dns v3.1.0

### OS Version
10.0.15063.413

### Description
If you have two DNS Zones with similar names i.e.(www.example.com and example.com) and you try to do a simple filter of DNS Zones using the where-object cmdlet, the expected result of returning the only one name is not achieved, instead you get both names, this is the case no matter which case you filter for, this can also be seen for the etag property of the DNSZone Object.

### Script/Steps for Reproduction
Get-AzureRmDnsZone | Where-Object {$_.Name -eq "example.com"}

**Comment by cormacpayne** [comment]
@animetauren Hey Henry, thanks for reporting this. We will have a PR out shortly that will fix this issue (just a matter of using the `enumerateCollection` parameter in [this `WriteObject` call](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Dns/Commands.Dns/Zones/GetAzureDnsZone.cs#L54)).

As a workaround until we get this fix in the next release next Friday, you can do

```cs
(Get-AzureRmDnsZone) | Where-Object { $_.Name -eq "example.com }
```

**Comment by animetauren** [comment]
@cormacpayne Thanks for the quick update and that fix does the job!

**Comment by cormacpayne** [comment]
@animetauren just opened PR https://github.com/Azure/azure-powershell/pull/4213 to fix this issue

This will enable the scenario you originally mentioned:

```powershell
Get-AzureRmDnsZone | Where-Object { $_.Name -eq "example.com" }
```

**Comment by cormacpayne** [comment]
@animetauren Hey Henry, this has been fixed in [version 4.2.0](https://github.com/Azure/azure-powershell/releases/tag/v4.2.0-July2017) of Azure PowerShell. Please let me know if you are still seeing this issue!
